src/makemsix/main.cpp

```diff
@@ -594,6 +628,52 @@ Command CreatePackCommand()
     return result;
 }
 
+Command CreateSignCommand()
+{
+    Command result{ "sign", "Signs an existing package in-place",
+        {
+            Option{ "-p", "Package file path.", true, 1, "package" },
+            Option{ "-c", "Certificate file path.", true, 1, "cert" },
+            Option{ "-cf", "Certificate format.", false, 1, "format" },
+            Option{ "-pass", "Certificat epassword.", false, 1, "password" },
+            // TODO: Potentially support other types of certificate files, along with separate private keys.
+            // TODO: Support passing in the certificate chain separately.
+            // TODO: Windows signing allows choosing whether to validate the block hashes, could add here.
+            // TODO: Potentially allow non-inplace by making an output file param.
+            // TODO: Full package content hash is optional
+            // TODO: Flag to control CI catalog generation
+            Option{ TOOL_HELP_COMMAND_STRING, "Displays this help text." },
+        }
+    };
+    result.SetDescription({
+        "Signs the given <package> file, modifying it to either include or replace a",
+        "signature. The <cert> file contains the signing certificate.",
+        "The following certificate formats are supported, and the format will be",
+        "inferred from the file name if not provided:",
+        "    pfx: [*.pfx] A PKCS12 containing both public and private keys",
+        "",
+        "WARNING: This feature is not yet complete. It has only had manual testing",
+        "         and does not yet allow for verification of custom certificates",
+        "         used during signing. It also required building with openssl."
+        });
+
+    result.SetInvocationFunc([](const Invocation& invocation)
+        {
+            std::cout << "WARNING: The signing feature is not complete, see the help for this command for more information." << std::endl;
+            std::cout << std::endl;
+
+            return SignPackage(
+                MSIX_SIGNING_OPTIONS::MSIX_SIGNING_OPTIONS_NONE,
+                const_cast<char*>(invocation.GetOptionValue("-p").c_str()),
+                GetCertificateFormat(invocation),
+                const_cast<char*>(invocation.GetOptionValue("-c").c_str()),
+                const_cast<char*>(invocation.GetOptionValue("-pass").c_str()),
```

Comment:
The OpenSSL docs specify the encoding of the password, but our `main` does not attempt to enforce any specific type of encoding on the input.  This would need to be addressed for full support.

Reply:
While that _is_ true, I think that the issue exists more broadly in consistently supporting various encodings in the command-line flags in makemsix. Addressing that fits better in the scope of a follow-up PR to handle it for both the case of passwords, as here, and also for the variety of cases that come up with filenames across OSes.